src/Spelling/WindowsSpellChecker.cs

```diff
@@ -0,0 +1,752 @@
+                internal SpellerStates sstate;
+                internal uint ichStart;
+                internal UIntPtr CchUse;
+            }
+
+            [StructLayout(LayoutKind.Sequential)]
+            internal struct Wsrb
+            {
+                internal unsafe char* pwsz;
+                internal unsafe SpellerSuggestion* prgsugg;
+                internal uint ichError;
+                internal uint CchError;
+                internal uint ichProcess;
+                internal uint CchProcess;
+                internal SpellerStatus sstat;
+                internal uint csz;
+                internal uint cszAlloc;
+                internal uint CchMac;
+                internal uint CchAlloc;
+            }
+
+            internal sealed class Speller : IDisposable
+            {
+                private SpellerAddUdr _addUdr;
+
+                private SpellerCheck _check;
+
+                private SpellerClearUdr _clearUdr;
+
+                private ProofCloseLex _closeLex;
+
+                private SpellerDelUdr _deleteUdr;
+
+                private IntPtr[] _lexicons;
+
+                private IntPtr _id;
+                private IntPtr _libraryHandle;
+                private IntPtr _ignoredDictionary;
+
+                private ProofOpenLex _openLex;
+
+                private ProofTerminate _terminate;
+
+                internal Speller(string path)
+                {
+                    IntPtr ptr;
+                    _libraryHandle = KernalNativeMethods.LoadLibrary(path);
+                    if (_libraryHandle == IntPtr.Zero)
+                    {
+                        throw new Win32Exception();
+                    }
+
+                    ProofInit proc = GetProc<ProofInit>(_libraryHandle, "SpellerInit");
+                    ProofSetOptions proofSetOptions = GetProc<ProofSetOptions>(_libraryHandle, "SpellerSetOptions");
+                    _terminate = GetProc<ProofTerminate>(_libraryHandle, "SpellerTerminate");
+                    _openLex = GetProc<ProofOpenLex>(_libraryHandle, "SpellerOpenLex");
+                    _closeLex = GetProc<ProofCloseLex>(_libraryHandle, "SpellerCloseLex");
+                    _check = GetProc<SpellerCheck>(_libraryHandle, "SpellerCheck");
+                    _addUdr = GetProc<SpellerAddUdr>(_libraryHandle, "SpellerAddUdr");
+                    _deleteUdr = GetProc<SpellerDelUdr>(_libraryHandle, "SpellerDelUdr");
+                    _clearUdr = GetProc<SpellerClearUdr>(_libraryHandle, "SpellerClearUdr");
+                    ProofParams pxpar = new ProofParams {VersionApi = 0x3000000};
+                    CheckErrorCode(proc(out ptr, ref pxpar));
+                    _id = ptr;
+                    CheckErrorCode(proofSetOptions(ptr, 0, 0x20006));
+                    InitIgnoreDictionary();
+                }
+
+                // Needed to clean-up unmanaged resources
+#pragma warning disable CA1821 // Remove empty Finalizers
+                ~Speller()
```

Comment:
:memo: Remove finalizer after moving to safe handles

Reply:
No longer applies